ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -603,23 +603,25 @@ private class PostgresLedgerDao(
     }.toMap
 
     dbDispatcher
-      .executeSql { implicit conn =>
-        // First, store all ledger entries without updating the ACS
-        // We can't use the storeLedgerEntry(), as that one does update the ACS
-        ledgerEntries.foreach {
-          case (i, le) =>
-            le match {
-              case tx: LedgerEntry.Transaction => storeTransaction(i, tx)
-              case rj: LedgerEntry.Rejection => storeRejection(i, rj)
-              case cp: LedgerEntry.Checkpoint => storeCheckpoint(i, cp)
-            }
-        }
+      .executeSql(
+        s"store initial state from scenario [${activeContracts.size} contracts, ${ledgerEntries.size} ledger entries") {
```

Comment:
Probably not too pressing, but it would probably better to force `activeContracts` and `ledgerEntries` to be `IndexedSeq`s to rule off calls to `size`s that could be linear.

Reply:
I'm not worried about this particular case, because it only happens on startup if you happen to run a scenario (i.e. non-production and/or single time occurrence only).